scripts/components/App.js

```diff
@@ -3,34 +3,88 @@ import React from "react";
 
 export class Form extends React.Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {record: this.props.record || {}};
+  }
+
   onFormSubmit(event) {
     event.preventDefault();
-    var record = {label: event.target.elements.label.value};
-    this.props.updateRecord(record);
+    var action = this.props.record ? 'update' : 'create';
+    this.props.events.emit('action:' + action, this.state.record);
+  }
+
+  onChange(field, event) {
+    var newrecord = Object.assign({}, this.state.record, {[field]: event.target.value});
+    this.setState({record: newrecord});
   }
 
   render() {
     return (
       <form onSubmit={this.onFormSubmit.bind(this)}>
-        <input name="label" type="text" />
-        <button type="submit">Add</button>
+        <input ref="label"
+               autofocus name="label" type="text"
+               placeholder="Label"
+               value={this.state.record.label}
+               onChange={this.onChange.bind(this, "label")} />
+        <button type="submit">{this.props.record ? "Save" : "Add"}</button>
       </form>
     );
   }
 }
 
 
+export class Item extends React.Component {
+  onEdit() {
+    this.props.events.emit('action:edit', this.props.index)
+  }
+
+  onDelete() {
+    this.props.events.emit('action:delete', this.props.item);
+  }
+
+  render() {
+    if (!this.props.editing) {
+      return (
+        <li key={this.props.index} onClick={this.onEdit.bind(this)}>{this.props.item.label}</li>
+      );
+    }
+    return (
+      <li key={this.props.index}>
+        <Form record={this.props.item} events={this.props.events}/>
+        <button onClick={this.onDelete.bind(this)}>Delete</button>
+      </li>
+    );
+  }
+}
+
+
 export class List extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {current: null};
+  }
 
-  static get defaultProps() {
-    return {items: []};
+  componentDidMount() {
+    this.props.events.on("action:edit", (index) => {
+      this.setState({current: index});
+    });
```

Comment:
Nit:

```js
this.props.events.on("action:edit", current => this.setState({current: current}));
this.props.events.on("action:update", _ => this.setState({current: null}));
this.props.events.on("action:delete", _ => this.setState({current: null}));
```

I wish we could subscribe to multiple events at once like with Backbone.Events… Oh well.